app/models/user.rb

```diff
@@ -19,4 +24,27 @@ def normalize_queue_item_positions
   def queue_item_in_queue?(queue_item)
     queue_items.include?(queue_item)
   end
+
+  def already_follows?(user)
+    following_relationships.map(&:leader).include?(user)
+  end
+
+  def can_follow?(user)
+    !(self.already_follows?(user) || user == self)
+  end
+
+  def create_reset_token
+    reset_token = SecureRandom.urlsafe_base64
+    update_attribute(:reset_token, reset_token)
+  end
+
+  def clear_reset_token
+    update_attribute(:reset_token, nil)
+  end
+
+  private
+
+  def generate_token
```

Comment:
Maybe you should have two methods, `set_token` and `generate_token`. `set_token` could be used for the `before_action`, and `generate_token` could be used by `create_reset_token` and `set_token`. That would keep your code generation in one place, which makes it easy to change your logic and keep everything working as you expect.